samples/tgidemo.c

```diff
@@ -12,8 +12,8 @@
 #  define DYN_DRV       1
 #endif
 
-#define COLOR_BACK      TGI_COLOR_BLACK
-#define COLOR_FORE      TGI_COLOR_WHITE
+#define COLOR_BACK      0
```

Comment:
Why?

Reply:
Oops, I see what you mean now; I misread the docs. Sorry!

Reply:
If I don't misunderstand the issue here then it's a variant of the issue we "always" have:

TGI_COLOR_... macros don't describe color values to be set into the palette. They describe colors in the default palette. There may be targets where those macros can be used for both purposes, but if they can't, then describing colors in the default palette takes precedence. The reasoning behind that is that TGI is meant to be a cross target API. And there are targets without palette in the first place so setting entries in the palette is already to be considered target specific code.

The user docs describe this pretty clearly: _The actual color is an index to a palette. During tgi_init you will get a default palette. The number of colors depend on the platform. All platforms recognize at least TGI_COLOR_BLACK and TGI_COLOR_WHITE. But some platforms have many more predefined colors. If you paint using TGI_COLOR_GREEN and then you change the green of the palette to blue using tgi_setpalette then after this painting in TGI_COLOR_GREEN will actually be blue._

I understand very well that it would be desirable to have a meaningful abstraction of the values to be set in the palette. This - and quite some other issues with the TGI API - have been discussed several times in the past decades. But there were no solutions because:
* there's no momentum to implement changes on all (or just the majority of) targets
* there are diametral opinions and the last significant discussion led to one member leaving the cc65 community

If TGI_COLOR_BLACK and TGI_COLOR_WHITE aren't defined as 0 and 1 then you need something like https://github.com/cc65/cc65/blob/master/libsrc/lynx/tgi_colors.s to "override" the default https://github.com/cc65/cc65/blob/master/libsrc/common/tgi_colors.s

Maybe somewhat interesting to read: https://github.com/cc65/cc65/pull/870